libs/sdk-ui-kit/src/List/ListItem.tsx

```diff
@@ -25,90 +25,144 @@ export interface ISingleSelectListItemProps {
-        super(props);
-        this.state = { isOverflowed: false };
-    }
+/**
+ * @internal
+ */
+export const SingleSelectListItem = forwardRef<
+    HTMLDivElement | HTMLButtonElement,
+    ISingleSelectListItemProps
+>(function SingleSelectListItem(props, ref) {
+    const {
+        title,
+        icon,
+        type,
+        className,
+        info,
+        eventsOnBubble = false,
+        hideDelayBubble,
+        isSelected,
+        isMenu,
+        elementType = "div",
+    } = props;
+
+    const [isOverflowed, setIsOverflowed] = useState(false);
+    const titleRef = useRef<HTMLSpanElement>(null);
+
+    const checkOverflow = useCallback(() => {
+        if (titleRef.current) {
+            const hasOverflow = titleRef.current.offsetWidth < titleRef.current.scrollWidth;
+            setIsOverflowed(hasOverflow);
+        }
+    }, []);
 
-    public componentDidMount(): void {
-        this.checkOverflow();
-    }
+    useEffect(() => {
+        checkOverflow();
 
-    public componentDidUpdate(): void {
-        this.checkOverflow();
-    }
+        window.addEventListener("resize", checkOverflow);
 
-    private checkOverflow(): void {
-        if (this.titleRef.current) {
-            // Checks if ellipsis has been applied on title span
-            const isOverflowed = this.titleRef.current.offsetWidth < this.titleRef.current.scrollWidth;
-            if (isOverflowed !== this.state.isOverflowed) {
-                // eslint-disable-next-line react/no-did-mount-set-state
-                this.setState({
-                    isOverflowed,
-                });
-            }
+        let resizeObserver: ResizeObserver | null = null;
+        if (titleRef.current && window.ResizeObserver) {
+            resizeObserver = new ResizeObserver(checkOverflow);
+            resizeObserver.observe(titleRef.current);
         }
-    }
 
-    private getClassNames = () => {
-        const { title, isSelected, isMenu, className } = this.props;
-        const generatedSeleniumClass = `s-${stringUtils.simplifyText(title)}`;
+        return () => {
+            window.removeEventListener("resize", checkOverflow);
+            if (resizeObserver) {
+                resizeObserver.disconnect();
+            }
+        };
+    }, [title, checkOverflow]);
 
+    const getClassNames = () => {
+        const generatedSeleniumClass = `s-${stringUtils.simplifyText(title)}`;
         return cx("gd-list-item", className, generatedSeleniumClass, {
             "is-selected": isSelected,
             "is-submenu": isMenu,
         });
     };
 
-    public render(): JSX.Element {
-        const { icon, onClick, onMouseOver, onMouseOut, type } = this.props;
-
-        if (type === "separator") {
-            return this.renderSeparatorItem();
-        }
-
-        if (type === "header") {
-            return this.renderHeaderItem();
-        }
+    const renderIcon = (iconProp: string | ReactNode) => {
```

Comment:
imho the type can be just `ReactNode`. `string` is its subtype.

Reply:
good catch